### PR TITLE
fix import and idempotency

### PR DESCRIPTION
**Pull request by shrsr**

### CODE REVIEW

File: mso/resource_mso_site.go [hunk 1]
```diff
@@ -113,49 +113,87 @@ func resourceMSOSite() *schema.Resource {
...
 	if err != nil {
 		return nil, err
 	}
 
-	d.SetId(models.StripQuotes(con.S("id").String()))
-
-	d.Set("name", models.StripQuotes(con.S("name").String()))
+	if platform == "nd" {
+		dataConAttr := con.S("common")
+		d.SetId(models.StripQuotes(con.S("id").String()))
```
Comment (lhercot) [inline mso/resource_mso_site.go:135]: Shouldn't this be outside of the if ? Right now it does not seems to apply to non-ND ones.

File: mso/resource_mso_site.go [hunk 2]
```diff
@@ -113,49 +113,87 @@ func resourceMSOSite() *schema.Resource {
...
+		d.Set("name", models.StripQuotes(dataConAttr.S("name").String()))
```
Comment (lhercot) [inline mso/resource_mso_site.go:136]: Shouldn't this be also in the non-ND ones (else) section?

File: mso/resource_mso_site.go [hunk 3]
```diff
@@ -113,49 +113,87 @@ func resourceMSOSite() *schema.Resource {
...
-	}
+		if dataConAttr.Exists("siteId") {
+			d.Set("apic_site_id", models.StripQuotes(dataConAttr.S("siteId").String()))
+		}
 
-	if con.Exists("password") {
-		d.Set("password", models.StripQuotes(con.S("password").String()))
-	}
+		if dataConAttr.Exists("labels") {
+			d.Set("apic_site_id", models.StripQuotes(dataConAttr.S("siteId").String()))
```
Comment (lhercot) [inline mso/resource_mso_site.go:143]: This should be labels not site_id right?

File: mso/resource_mso_site.go [hunk 4]
```diff
@@ -113,49 +113,87 @@ func resourceMSOSite() *schema.Resource {
...
+	if platform == "nd" {
+		dataConAttr := con.S("common")
+		d.SetId(models.StripQuotes(con.S("id").String()))
+		d.Set("name", models.StripQuotes(dataConAttr.S("name").String()))
 
-	if con.Exists("username") {
-		d.Set("username", models.StripQuotes(con.S("username").String()))
-	}
+		if dataConAttr.Exists("siteId") {
+			d.Set("apic_site_id", models.StripQuotes(dataConAttr.S("siteId").String()))
```
Comment (lhercot) [inline mso/resource_mso_site.go:139]: Where is the equivalent of this in the other one (else part)?

File: mso/resource_mso_site.go [hunk 5]
```diff
@@ -363,47 +401,71 @@ func resourceMSOSiteRead(d *schema.ResourceData, m interface{}) error {
...
-			d.Set("username", matches[2])
-			d.Set("login_domain", matches[1])
+
+	if platform == "nd" {
+		dataConAttr := con.S("common")
+		d.SetId(models.StripQuotes(con.S("id").String()))
+		d.Set("name", models.StripQuotes(dataConAttr.S("name").String()))
+
+		if dataConAttr.Exists("siteId") {
+			d.Set("apic_site_id", models.StripQuotes(dataConAttr.S("siteId").String()))
```
Comment (lhercot) [inline mso/resource_mso_site.go:411]: Look at issues listed above for this section too.

File: mso/resource_mso_site.go [hunk 6]
```diff
@@ -363,47 +401,71 @@ func resourceMSOSiteRead(d *schema.ResourceData, m interface{}) error {
...
-	}
-	if con.Exists("location") {
-		loc1 := con.S("location").Data()
-		locset := make(map[string]interface{})
-		if loc1 != nil {
-			loc := loc1.(map[string]interface{})
-			locset["lat"] = fmt.Sprintf("%v", loc["lat"])
-			locset["long"] = fmt.Sprintf("%v", loc["long"])
+		if dataConAttr.Exists("labels") {
+			d.Set("apic_site_id", models.StripQuotes(dataConAttr.S("siteId").String()))
```
Comment (lhercot) [inline mso/resource_mso_site.go:415]: Shouldn't this be labels?

**Review by lhercot — CHANGES_REQUESTED** [review]

File: mso/resource_mso_site.go (hunk 2 above)
Comment (shrsr) [reply mso/resource_mso_site.go:137]: Yes. Made a fix.

File: mso/resource_mso_site.go (hunk 3 above)
Comment (shrsr) [reply mso/resource_mso_site.go]: Yes. I switched it to labels in the latest commit.

File: mso/resource_mso_site.go (hunk 4 above)
Comment (shrsr) [reply mso/resource_mso_site.go:140]: Fixed in the latest commit.

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**